app/mailers/asset_notification.rb

```diff
@@ -1,28 +1,34 @@
 class AssetNotification < ActionMailer::Base
   default from: Alonetone.email
 
-  def upload_notification(asset, email, _sent_at = Time.now)
-    @track = asset.name
-    @description = asset.description
-    @name = asset.user.name
-    @user = asset.user
-    @title = asset.title.present? ? asset.title : "new track"
-    @play_link = play_link_for(asset)
-    @user_link = user_link_for(asset)
+  def upload_notification(assets, user, _sent_at = Time.now)
+    @user = user
+    @tracks = generate_track_hash(assets)
     @stop_following_link = stop_following_link
     @unsubscribe_link = unsubscribe_link
     @exclamation = %w[Sweet Yes Oooooh Alright Booya Yum Celebrate OMG].sample
-    mail subject: "[alonetone] '#{asset.user.name}' uploaded a new track!", to: email
+    @upload_user_url = user_link_for(assets.first)
+    mail subject: "[alonetone] '#{@user.name}' uploaded a new track!", to: user.email
   end
 
   protected
 
+  # I'm sure there is some fancy shmancy way to do the same thing
+  # but I can't think of it right now.
+  def generate_track_hash(assets)
+    assets_array = []
+    assets.each do |asset|
+      assets_array << { title: asset.title, play_link: play_link_for(asset) } if asset.title
+    end
+    assets_array
+  end
+
   def user_link_for(asset)
     'https://' + Alonetone.url + '/' + asset.user.login
   end
 
   def play_link_for(asset)
-    user_link_for(asset) + '/tracks/' + asset.id.to_s
+    user_link_for(asset) + '/tracks/' + asset.permalink.to_s
```

Comment:
This was using `id` for a reason, see my other comment :)

Reply:
Yep, reverting.